hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/BaseLoadBalancer.java

```diff
@@ -870,6 +870,9 @@ void regionMoved(int region, int oldServer, int newServer) {
         if (regions[i] == regionIndex) {
           break;
         }
+        if (i == regions.length - 1) {
+            return Arrays.copyOf(regions, regions.length);
```

Comment:
Here i think we can just return regions. No need to copy the entire array, unless you have a strong reason.